backend/apps/project_health/services.py

```
@@ -0,0 +1,216 @@
+        elif close_ratio < 0.3:
+            score -= 10
+            red_flags.append(f"🔴 Low issue resolution rate ({int(close_ratio * 100)}% closed).")
+
+    # Contributor diversity
+    contributors = data.get("contributor_count", 0)
+    if contributors > 50:
+        score += 10
+        green_flags.append(f"✅ Large contributor community ({contributors} contributors).")
+    elif contributors < 5:
+        score -= 5
+        red_flags.append("🟡 Low contributor count – bus-factor risk.")
```

Ruff (0.15.21)

[warning] String contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF001)